tests/test_spidermon_signal_connect.py

```diff
@@ -14,34 +14,35 @@ def get_crawler(settings_dict=None):
 
 
 @pytest.mark.skipif(sys.version_info < (3, 4), reason="requires python3.4 or higher")
-def test_spider_opened_connect_signal():
-    from unittest import mock
+def test_spider_opened_connect_signal(mocker):
 
-    with mock.patch.object(Spidermon, "spider_opened") as spider_opened_method:
-        crawler = get_crawler(
-            settings_dict={
-                "SPIDERMON_ENABLED": True,
-                "EXTENSIONS": {"spidermon.contrib.scrapy.extensions.Spidermon": 100},
-            }
-        )
-        spider = Spider.from_crawler(crawler, "example.com")
-        crawler.signals.send_catch_log(signal=signals.spider_opened, spider=spider)
-        assert spider_opened_method.called, "spider_opened not called"
+    spider_opened_method = mocker.patch.object(Spidermon, "spider_opened")
```

Comment:
`autospec=True` will do the trick.
Don't forget to specify it everywhere or mock.asser_called_with_nonce and such funny staff will pass.